### PR TITLE
XSI-1773 clean up swtpm service files

### PR DESCRIPTION
**Pull request by lindig**
We have seen swtpm systemd service files not being removed. We now make Fe_systemctl.stop callable when the servive is potentially not running to ensure clean up is happening regardless.

### CODE REVIEW

**Review by psafont — APPROVED** [review]
`Fe_systemctl.stop` unlink the service file, if it exists.

**Comment by psafont** [comment]
This might not be enough for all cases, but probably covers most of them. Do we need to clean up before creating a service?

File: ocaml/xenopsd/xc/service.ml [hunk 1]
```diff
@@ -619,15 +619,14 @@ module SystemdDaemonMgmt (D : DAEMONPIDPATH) = struct
     match (of_domid domid, is_running ~xs domid) with
```
Comment (robhoes) [inline ocaml/xenopsd/xc/service.ml:619]: This (existing code) is makes some unnecessary calls, as `is_running` also calls `of_domid` internally. And now we don't even care about the result of `is_running` in the systemd case. So better would be to drop `is_running` from the match and call `Compat.is_running ~xs domid` in the `None` case below.

**Review by robhoes — APPROVED** [review]
I think it's a good fix, but we can easily make it more efficient too.

**Review by robhoes — APPROVED** [review]

**Pull request merged**